File: solutions/2025-12-21/solution.py
# Longest Consecutive Sequence
# Difficulty: Medium
# URL: https://leetcode.com/problems/longest-consecutive-sequence/

class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0
        nums = set(nums)
        max_length = 0
        for num in nums:
            if num - 1 not in nums:
                current_num = num
                current_length = 1
                while current_num + 1 in nums:
                    current_num += 1
                    current_length += 1
                max_length = max(max_length, current_length)
        return max_length

File: solutions/2025-12-21/solution.java
// Longest Consecutive Sequence
// Difficulty: Medium
// URL: https://leetcode.com/problems/longest-consecutive-sequence/

class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums.length == 0) {
            return 0;
        }
        Set<Integer> set = new HashSet<>();
        for (int num : nums) {
            set.add(num);
        }
        int max_length = 0;
        for (int num : set) {
            if (!set.contains(num - 1)) {
                int current_num = num;
                int current_length = 1;
                while (set.contains(current_num + 1)) {
                    current_num += 1;
                    current_length += 1;
                }
                max_length = Math.max(max_length, current_length);
            }
        }
        return max_length;
    }
}